src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionString.cs

```diff
@@ -0,0 +1,337 @@
+﻿namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// This class represents the connection string being set by the user
+    /// e.g. TEST_CSM_ORGID_AUTHENTICATION="AADTenant=72f98AAD-86f1-2d7cd011db47;ServicePrincipal=72f98AAD-86f1-2d7cd011db47;Password=tzT2+LJBRkSAursui7/Qgo+hyQQ=;SubscriptionId=5562fbd2-HHHH-WWWW-a55d-lkjsldkjf;BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
+    /// </summary>
+    public class ConnectionString
+    {
+        #region fields
+        Dictionary<string, string> _keyValuePairs;
+        string _connString;
+        StringBuilder _parseErrorSb;
+        #endregion
+
+        #region Properties
+
+        private bool CheckViolation { get; set; }
+
+        
+
+        /// <summary>
+        /// Represents key values pairs for the parsed connection string
+        /// </summary>
+        public Dictionary<string,string> KeyValuePairs
+        {
+            get
+            {
+                if(_keyValuePairs == null)
+                {
+                    _keyValuePairs = new Dictionary<string, string>();
+                }
+                return _keyValuePairs;
+            }
+            private set
+            {
+                _keyValuePairs = value;
+            }
+        }
+        
+        /// <summary>
+        /// Returns all the parse errors while parsing connection string
+        /// </summary>
+        public string ParseErrors
+        {
+            get
+            {
+                if(_parseErrorSb == null)
+                {
+                    _parseErrorSb = new StringBuilder();                    
+                }
+                return _parseErrorSb.ToString();
+            }
+
+            private set
+            {
+                if (_parseErrorSb == null)
+                {
+                    _parseErrorSb = new StringBuilder();
+                    _parseErrorSb.AppendLine(value);
+                }
+                else
+                    _parseErrorSb.AppendLine(value);
+            }
+        }
+
+        #endregion
+
+        #region Constructor/Init
+        void Init()
+        {
+            List<string> connectionKeyNames = (from fi in typeof(ConnectionStringKeys).GetFields(BindingFlags.Public | BindingFlags.Static) select fi.GetRawConstantValue().ToString()).ToList<string>();
```

Comment:
please limit line length to about 100 characters. Otherwise it is hard to do code review on github through browser.